gui/src/2D/controls/toggleButton.ts

```diff
@@ -0,0 +1,310 @@
+     */
+    public pointerUpActiveAnimation: () => void;
+
+    /**
+     * Function called to generate a pointer enter animation when the toggle button is inactive.
+     */
+    public pointerEnterInactiveAnimation: () => void;
+    /**
+     * Function called to generate a pointer out animation when the toggle button is inactive.
+     */
+    public pointerOutInactiveAnimation: () => void;
+    /**
+     * Function called to generate a pointer down animation when the toggle button is inactive.
+     */
+    public pointerDownInactiveAnimation: () => void;
+    /**
+     * Function called to generate a pointer up animation when the toggle button is inactive.
+     */
+    public pointerUpInactiveAnimation: () => void;
+
+    /** Observable raised when isActive is changed */
+    public onIsActiveChangedObservable = new Observable<boolean>();
+
+    /**
+     * Gets or sets a boolean indicating that the toggle button will let internal controls handle picking instead of doing it directly using its bounding info
+     */
+    public delegatePickingToChildren = false;
+
+    private _image: Nullable<Image>;
+    /**
+     * Returns the ToggleButton's image control if it exists
+     */
+    public get image(): Nullable<Image> {
+        return this._image;
+    }
+
+    private _textBlock: Nullable<TextBlock>;
+    /**
+     * Returns the ToggleButton's child TextBlock control if it exists
+     */
+    public get textBlock(): Nullable<TextBlock> {
+        return this._textBlock;
+    }
+
+    private _group: string;
+    /** Gets or sets group name this toggle button belongs to */
+    public get group(): string {
+        return this._group;
+    }
+    public set group(value: string) {
+        if (this._group === value) {
+            return;
+        }
+
+        this._group = value;
+    }
+
+    private _isActive = false;
+    /** Gets or sets a boolean indicating if the toogle button is active or not */
+    public get isActive(): boolean {
+        return this._isActive;
+    }
+    public set isActive(value: boolean) {
+        // Function modeled after radioButton.ts
+        if (this._isActive === value) {
+            return;
+        }
+
+        this._isActive = value;
+
+        // Update the visual state based on the new value
+        if (this._isActive) {
+            this.toActiveAnimation?.();
+        } else {
+            this.toInactiveAnimation?.();
+        }
+
+        this._markAsDirty();
+
+        this.onIsActiveChangedObservable.notifyObservers(value);
+
+        if (this._isActive && this._host && this._group) {
+            // A toggle button in a group can only have 1 active element at a given time.
+            // If this toggle button has a group, set other toggle buttons in the group to inactive.
+            this._host.executeOnAllControls((control) => {
+                // Check for control type ToggleButton
+                if ((<any>control)._getTypeName === "ToggleButton") {
```

Comment:
Something like this should work:
```javacsript
if (control.typeName === "ToggleButton") {
```
as `typeName` is a property of the base class `Control`.

Reply:
Ah, that feels much cleaner. This pull request has been updated with this change.